libs/ml/examples/commodity_prediction/main.cpp

```diff
@@ -0,0 +1,349 @@
+  SizeType    output_feature_size = 0;
+
+  /// READ ARGUMENTS
+  if ((i = ArgPos((char *)"-model_num", argc, argv)) > 0)
+  {
+    model_num = static_cast<SizeType>(std::stoul(argv[i + 1]));
+  }
+  if ((i = ArgPos((char *)"-input_dir", argc, argv)) > 0)
+  {
+    input_dir = argv[i + 1];
+  }
+
+  std::string architecture_file = input_dir + "/architecture.csv";
+
+  /// DEFINE NEURAL NET ARCHITECTURE ///
+
+  auto g_ptr = std::make_shared<fetch::ml::Graph<ArrayType>>(GraphType());
+
+  auto                     arch_tuple = read_architecture(architecture_file, g_ptr, model_num);
+  std::string              dataname   = arch_tuple.first;
+  std::vector<std::string> node_names = arch_tuple.second;
+
+  std::string test_x_file = input_dir + "/" + dataname + "_x_test.csv";
+  std::string test_y_file = input_dir + "/" + dataname + "_y_pred_test.csv";
+  std::string weights_dir = input_dir + "/output/" + dataname + "/model_weights";
+
+  /// LOAD WEIGHTS INTO GRAPH ///
+  auto sd = g_ptr->StateDict();
+
+  for (auto const &name : node_names)
+  {
+    if (name.find("dense") != std::string::npos)
+    {
+      // if it is a dense layer there will be weights and bias files
+      std::vector<std::string> dir_list =
+          fetch::ml::examples::GetAllTextFiles(weights_dir + "/" + name, "");
+      std::vector<std::string> actual_dirs;
+      for (auto dir : dir_list)
+      {
+        if (dir != "." && dir != "..")
+        {
+          actual_dirs.emplace_back(dir);
+        }
+      }
+      assert(actual_dirs.size() == 1);
+
+      std::string node_weights_dir = weights_dir + "/" + name + "/" + actual_dirs[0];
+      ArrayType   weights          = read_csv(node_weights_dir + "/kernel:0.csv", 0, 0, true);
+      ArrayType   bias             = read_csv(node_weights_dir + "/bias:0.csv", 0, 0, false);
+
+      assert(bias.shape()[0] == weights.shape()[0]);
+
+      auto weights_tensor = sd.dict_[name + "_FC_Weights"].weights_;
+      auto bias_tensor    = sd.dict_[name + "_FC_Bias"].weights_;
+
+      *weights_tensor     = weights;
+      *bias_tensor        = bias;
+      output_feature_size = weights.shape()[0];  // stores shape of last dense layer
+    }
+  }
+
+  // load state dict into graph (i.e. load pretrained weights)
+  g_ptr->LoadStateDict(sd);
+
+  /// FORWARD PASS PREDICTIONS ///
+  ArrayType test_x           = read_csv(test_x_file, 1, 1);
+  ArrayType test_y           = read_csv(test_y_file, 1, 1);
+  SizeType  output_data_size = test_y.shape()[0];
+  assert(output_feature_size == test_y.shape()[1]);
+  assert(test_y.shape()[0] == test_x.shape()[0]);
+
+  ArrayType output({output_data_size, output_feature_size});
+
+  for (SizeType j = 0; j < output_data_size; j++)
+  {
+    auto current_input = test_x.Slice(j).Copy().Transpose();
+    g_ptr->SetInput("num_input", current_input);
+    auto slice_output = g_ptr->Evaluate(node_names.back());
+    output.Slice(j).Assign(slice_output);
+  }
+
+  assert(output.AllClose(test_y, 0.00001f));
```

Comment:
instead of asserting just print a statement indicating success or failure (otherwise running this example in release mode does nothing interesting)